packages/jellyfish-api-core/__tests__/category/loan/takeLoan.test.ts

```diff
@@ -0,0 +1,216 @@
+
+    await tGroup.get(0).rpc.loan.depositToVault({
+      id: vaultId, from: collateralAddress, amount: '1@BTC'
+    })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+  }
+
+  it('should not takeLoan on nonexistent vault', async () => {
+    const promise = tGroup.get(0).rpc.loan.takeLoan({
+      vaultId: '0'.repeat(64),
+      amounts: '30@TSLA'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow(`Vault <${'0'.repeat(64)}> not found`)
+  })
+
+  it('should not takeLoan on nonexistent loan token', async () => {
+    const promise = tGroup.get(0).rpc.loan.takeLoan({
+      vaultId: vaultId,
+      amounts: '1@BTC'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('Loan token with id (1) does not exist!')
+  })
+
+  it('should not takeLoan on invalid token', async () => {
+    const promise = tGroup.get(0).rpc.loan.takeLoan({
+      vaultId: vaultId,
+      amounts: '1@INVALID'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('Invalid Defi token: INVALID')
+  })
+
+  it('should not takeLoan on incorrect auth', async () => {
+    const promise = tGroup.get(1).rpc.loan.takeLoan({
+      vaultId: vaultId,
+      amounts: '30@TSLA'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow(`Incorrect authorization for ${vaultAddress}`)
+  })
+
+  it('should not takeLoan while exceed vault collateralization ratio', async () => {
+    const promise = tGroup.get(0).rpc.loan.takeLoan({
+      vaultId: vaultId,
+      amounts: '300000@TSLA'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('Vault does not have enough collateralization ratio defined by loan scheme')
+  })
+
+  it('should not takeLoan on mintable:false token', async () => {
+    await tGroup.get(0).container.call('updateloantoken', ['TSLA', { mintable: false }])
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+
+    const promise = tGroup.get(0).rpc.loan.takeLoan({
+      vaultId: vaultId,
+      amounts: '30@TSLA'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('Loan cannot be taken on token with id (2) as "mintable" is currently false')
+
+    await tGroup.get(0).container.call('updateloantoken', ['TSLA', { mintable: true }])
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+  })
+
+  it('should takeLoan', async () => {
+    const vaultBefore = await tGroup.get(0).container.call('getvault', [vaultId])
+    const vaultBeforeTSLAAcc = vaultBefore.loanAmount.length > 0
+      ? vaultBefore.loanAmount.find((amt: string) => amt.split('@')[1] === 'TSLA')
+      : undefined
+    const vaultBeforeTSLAAmt = vaultBeforeTSLAAcc !== undefined ? Number(vaultBeforeTSLAAcc.split('@')[0]) : 0
```

Comment:
Is `vaultBeforeTSLAAmt` will always be zero?  If yes, line 171 to 175 should be removed.

Reply:
it uses for assert the changes of 
```
expect(vaultAfterTSLAAmt - vaultBeforeTSLAAmt).toStrictEqual(40.00002280)
```